src/nvtxw.rs

```diff
@@ -0,0 +1,403 @@
+                        schemaId: LEGION_NVTXW_PAYLOAD_SCHEMA_ID,
+                        size: size_of::<legion_nvtxw_event>(),
+                        payload: events.as_ptr() as *const c_void,
+                    },
+                ];
+
+                nvtxw::event_write(interface, stream, &payloads).expect("Failed to write event");
+            }
+        }
+    }
+
+    fn process_events(
+        data_source: &mut CountingDeferredDataSource<T>,
+        interface: nvtxw::InterfaceHandle,
+        streams: &BTreeMap<EntryID, nvtxw::StreamHandle>,
+        zero_time: i64,
+        unmatched_tiles: &mut UnmatchedTileHold,
+        num_requests: u64,
+    ) {
+        while data_source.outstanding_requests() > num_requests {
+            // let summary_tiles = data_source.get_summary_tiles();
+            let slot_tiles = data_source.get_slot_tiles();
+            let slot_meta_tiles = data_source.get_slot_meta_tiles();
+
+            for tile in slot_tiles {
+                let e = tile.entry_id.clone();
+                unmatched_tiles.entry(e).or_insert((None, None)).0 = Some(tile);
+            }
+
+            for meta_tile in slot_meta_tiles {
+                let e = meta_tile.entry_id.clone();
+                unmatched_tiles.entry(e).or_insert((None, None)).1 = Some(meta_tile);
+            }
+
+            unmatched_tiles.retain(|_entry_id, (ut, um)| {
+                if let (Some(tile), Some(meta_tile)) = (ut, um) {
+                    Self::write_matched_tile(interface, streams, zero_time, tile, meta_tile);
+                    return false;
+                }
+                true
+            });
+        }
+    }
+
+    pub fn write(mut self) -> io::Result<()> {
+        self.data_source.fetch_info();
+        let mut info = None;
+        while info.is_none() {
+            info = self.check_info();
+        }
+        let info = info.unwrap();
+
+        let entry_ids = walk_entry_list(&info.entry_info);
+
+        let full_range_tile_id = TileID(info.interval);
+        let full = true;
+
+        // For now, this only works on dynamic data sources
+        assert!(info.tile_set.tiles.is_empty());
+
+        println!("Exporting to NVTXW");
+
+        let interface = nvtxw::initialize_simple().expect("Failed to initialize NVTXW");
+
+        let session = nvtxw::session_begin_simple(interface, self.output, self.merge)
+            .expect("Failed to create session");
+
+        let c_event_name = CString::new("Legion Event").expect("CString::new failed");
+
+        let c_field_name_time_start = CString::new("time_start").expect("CString::new failed");
+        let c_field_name_time_stop = CString::new("time_stop").expect("CString::new failed");
+        let c_field_name_name = CString::new("name").expect("CString::new failed");
+        let c_field_name_color = CString::new("color").expect("CString::new failed");
+
+        // The C string containing the event's name must be provided in a special way.
```

Comment:
Let's either explain what the "special way" is, or else remove this comment if it's not necessary. As it is this is too vague to be helpful.

Reply:
I changed the comment.